testsuite: Crashing packages for openSUSE:Leap:15

We had opensuselp15 test target broken for few months. Now it's been fixed and we are facing the following failing test:

```
[gw1] linux -- Python 3.6.10 /usr/bin/python3
capsys = <_pytest.capture.CaptureFixture object at 0x7f55c8dd8d68>
packages = [PosixPath('test/source/not-compressed-multi-spec-1.0-0.src.rpm'), PosixPath('test/source/CheckInclude-1-1.src.rpm'), ...'), PosixPath('test/source/non-utf8-1.0-1.el5.centos.src.rpm'), PosixPath('test/source/no-spec-file-0-0.src.rpm'), ...]
    @pytest.mark.parametrize('packages', [list(Path('test').glob('*/*.rpm'))])
    @pytest.mark.no_cover
    def test_run_full_rpm(capsys, packages):
        number_of_pkgs = len(packages)
        additional_options = {
            'rpmfile': packages,
        }
        options = {**options_preset, **additional_options}
        linter = Lint(options)
        linter.run()
        out, err = capsys.readouterr()
>       assert f'{number_of_pkgs} packages and 0 specfiles checked' in out
E       AssertionError: assert '93 packages and 0 specfiles checked' in '============================ rpmlint session starts ============================\nrpmlint: 0.0.0\nconfiguration:\n   ...sing /usr/bin/alternative-borked\n\n======== 86 packages and 0 specfiles checked; 464 errors, 1375 warnings ========\n'
```

It seems that some packages cause rpm to crash. Some of the crashes could be caused (or exposed) by pr#441 but as the leap15 target was broken for a few months, it's possible that some of them are completely unrelated.

--- a/rpmlint/checks/AlternativesCheck.py
+++ b/rpmlint/checks/AlternativesCheck.py
@@ -39,8 +39,8 @@ class AlternativesCheck(AbstractCheck):
             return
 
         # populate scriptlets
-        self.post = pkg.header[rpm.RPMTAG_POSTIN]
-        self.postun = pkg.header[rpm.RPMTAG_POSTUN]
+        self.post = byte_to_string(pkg.header[rpm.RPMTAG_POSTIN])
+        self.postun = byte_to_string(pkg.header[rpm.RPMTAG_POSTUN])
 
         if not self._check_ua_presence(pkg):
             return

--- a/test/spec/SpecCheck2.spec
+++ b/test/spec/SpecCheck2.spec
@@ -47,4 +47,3 @@ rm -rf $RPM_BUILD_ROOT
 %{_libdir}/foo
 
 %changelog
-Fake macros such as %%buildroot %!buildroot %.buildroot does not cause a warning.

